File: CivitasJava/src/civitas/CasillaSorpresa.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package civitas;

import java.util.ArrayList;

/**
 *
 * @author luis
 */
public class CasillaSorpresa extends Casilla{
    private MazoSorpresas mazo;
    private Sorpresa sorpresa;
    
    CasillaSorpresa(MazoSorpresas mazo, String nombre){
        super(nombre);
        this.mazo = mazo;
        sorpresa = null;
    }
    
    @Override
    void recibeJugador(int iactual, ArrayList<Jugador> todos){
        if(jugadorCorrecto(iactual,todos)){
            informe(iactual, todos);
            sorpresa = mazo.siguiente();
            sorpresa.aplicarAJugador(iactual, todos);   
        }
    }
    
    @Override
    public String toString(){
        String e = "Nombre: " + this.getNombre() + "\n";
        return e;
    }
}

File: CivitasRuby/lib/casilla_sorpresa.rb
# encoding:utf-8
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

module Civitas
  class CasillaSorpresa < Casilla
    
    def initialize(mazo, nombre)
        super(nombre)
        @mazo = mazo
    end
    
     def recibe_jugador(iactual, todos)
      if jugador_correcto(iactual,todos)
        informe(iactual,todos)
        @sorpresa = @mazo.siguiente();
        @sorpresa.aplicar_a_jugador(iactual, todos)
      end
    end
    
    def to_s
      "Nombre: #{@nombre} \n"
    end
    
  end
end